### DISCUSSION TITLE
InfluxDB 2  + Create Task

### DISCUSSION THREAD

**Discussion by nuiler** [ha-puzzles/evcc-grafana-dashboards#13] [Ideas]
Gibt es eine Möglichkeit den Script direkt auf InfluxDB unter den Tasks laufen zu lassen?
![image](https://github.com/ha-puzzles/evcc-grafana-dashboards/assets/16724275/2bbb1771-c94d-4598-bba5-a124eeb941bd)

**Comment by cschlipf** [comment]
Ehrlich gesagt, ich sehe das UI zum ersten Mal. Wie kommt man denn da drauf?

Aber das wäre ja echt cool, wenn das gehen würde.

**Comment by nuiler** [comment]
Ich habe InfluxDB v2.7.6 installiert und dann einen Zugriff v1 InfluxQL 
eingerichtet, damit ich ihre Vorlage verwenden kann.



------ Originalnachricht ------
Von "Carsten Schlipf" ***@***.***>
An "ha-puzzles/evcc-grafana-dashboards" 
***@***.***>
Cc "nuiler" ***@***.***>; "Author" 
***@***.***>
Datum 13.04.2024 09:14:49
Betreff Re: [ha-puzzles/evcc-grafana-dashboards] InfluxDB 2 + Create 
Task (Discussion #13)

**Reply by cschlipf** [reply]
Verstehe. Ich habe eine InfluxDB 1.8.x und dort scheint es dieses UI noch nicht zu geben. Daher kann ich hier leider nicht helfen. Aber wenn man hier Shell Scripte ausführen kann, dann sollte das in der Theorie funktionieren und wäre natürlich die eleganteste Lösung.

Falls es tut, bitte ich um Feedback.

**Comment by nuiler** [comment]
Ich bin da leider absoluter Anfänger. Aber so wie ich dass sehe, muss 
man Flux Scripte anlegen.

https://docs.influxdata.com/influxdb/cloud/process-data/get-started/

Vielleicht können Sie mir ein Beispiel zeigen, wie dies aussehen sollte.





------ Originalnachricht ------
Von "Carsten Schlipf" ***@***.***>
An "ha-puzzles/evcc-grafana-dashboards" 
***@***.***>
Cc "nuiler" ***@***.***>; "Author" 
***@***.***>
Datum 13.04.2024 17:38:59
Betreff Re: [ha-puzzles/evcc-grafana-dashboards] InfluxDB 2 + Create 
Task (Discussion #13)

**Reply by cschlipf** [reply]
Nein, da kann ich leider nicht helfen. Zudem mein eigenes Dashboard kein Flux supported und Flux bereits abgekündigt wurde.

**Comment by cschlipf** [comment]
Eine Möglichkeit das unabhängig von Shell Scripten zu machen, scheinen mir Continous Queries zu sein: https://docs.influxdata.com/influxdb/v1/query_language/continuous_queries/

Damit scheint man das alles auch ohne Shell Scripts direct in Influx machen zu können. Problem das ich hier sehe ist allerdings, dass auch das hier (wir auch das Script) sehr abhängig von der Version der InfluxDB ist. Im Fall eine Influx 2 sind das dann die Tasks, die dann aber unter dem abgesagten Flux laufen müssen, was von meinen Dashboards nicht unterstützt wird. 

Es ist echt schwierig :-(. Mit Flux müsste ich die ganze Arbeit nochmal neu machen und Flux ist eine Einbahnstraße. Ich tendiere dazu derzeit nur Influx 1.8 zu supporten und später auf Influx 3 zu gehen.

**Reply by cschlipf** [reply]
Habe mal in die continuous queries reingeschaut. Leider scheitert das bei InfluxQL daran, dass es keine 'today()' function gibt, die den. Zeitstempel des heutigen Tages um Mitternacht zurückgibt. Die bräuchte ich aber um mindestens die Daily-Aggregation zu machen (eine monatliche könnte ich dann in Grafana ableiten). Damit kann ich leider keine Where clause in continuous queries erzeugen, die auf den  aktuellen Tageszeitraum filtert :-( - das ist leider ein Deal-Breaker.

Eventuell geht das dann später mit Influx 3. In Influx 2 gibt es die today() Funktion mit Flux, aber wie schon gesagt, da Flux deprecated ist, werde ich damit nicht mehr anfangen. Und es bringt auch nichts für Home Assistant, wo üblicherweise eine Influx 1.8 installiert wird.

**Comment by cschlipf** [comment]
Ich schließe diese Diskussion mal. Siehe auch Issue #14

**Reply by Michel83** [reply]
Klar, kein Problem ich denke es funktioniert ja jetzt. Danke !
